feat(stack-view): initial design guidance (#23)
vmware-clarity/clarity-guidance: 1029-stack-view-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-21
---

# Stack View Component

> A [Stack View](https://clarity.design/documentation/stack-view) displays key/value pairs, which users can expand to show more detail.

Reviewer: ```suggestion
The [stack view](https://clarity.design/documentation/stack-view) component displays key/value pairs which users can expand to show more detail.
```

> The [stack view](https://clarity.design/documentation/stack-view) component displays key/value pairs, which users can expand to show more detail.

Reviewer 2: ```suggestion
The [stack view](https://clarity.design/documentation/stack-view) component displays key/value pairs that users can expand to reveal more details, providing a clear and organized way to present information.
```

## Guidance

> ### Stack View Usage

Reviewer: ```suggestion
### Usage
```

I don't think we need to repeat the component name. It's in the level-one heading. This is also consistent with the website. Please update all three level-three headings.

> - **Must** use stack view to display related key/value pairs.

Reviewer: ```suggestion
- **May** use stack view to display related key/value pairs.
```

There are valid use cases for key/value pairs without using a stack view.

> - **Must** use stack view to progressively display a large set of data (key/value pairs) or to enable users to reveal more complex or less frequently used data as needed.

Reviewer 2: ```suggestion
- Design teams **must** use the stack view to progressively display a large set of data (key/value pairs) or to allow users to reveal more complex or less frequently used data as needed, enhancing clarity and usability.
```

> - **May** use stack view to display related key/value pairs.

Reviewer 2: ```suggestion
Design teams **may** use the stack view to display related key/value pairs, helping users understand connections between the data while maintaining an organized layout.```

### Stack View Value Editing

> - **Must** have an edit button on the top-right of the stack view.

Reviewer: ```suggestion
- An editable stack view **must** have an edit button above the stack view to the right.
```

> - An editable stack view **must** have an edit button above the stack view to the right.

Reviewer 2: ```suggestion
Design teams **must** place an edit button above the editable stack view, positioned to the right, to provide users with easy access to editing options.```

> - An editable stack view **must** open in a modal on clicking the edit button.

Reviewer: ```suggestion
- An editable stack view **must** open a modal on clicking the edit button.
```

extra word here, sorry if I introduced this with my previous suggestion.

> - An editable stack view **must** open a modal on clicking the edit button.

Reviewer 2: ```suggestion
- Design teams **must** ensure that clicking the edit button on an editable stack view opens a modal, allowing users to make changes without losing context.
```

### Stack View Labels

> - **Must** provide coincise labels (keys) that let users know what is under the label.

Reviewer 2: ```suggestion
- Design teams **must** provide concise labels (keys) that clearly indicate what information is associated with each label, enhancing user understanding and navigation.
```

> - **Must** NOT use icons as a label in the stack view.

Reviewer: ```suggestion
- **Must not** use icons as a label in the stack view.
```

> - **Must not** use icons as a label in the stack view.

Reviewer 2: ```suggestion
- Design teams **must not** use icons as labels in the stack view, as this can lead to confusion and reduce clarity in understanding the associated data.
```
